beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/CommonAncestor.java

```diff
@@ -24,63 +26,107 @@
 
 public class CommonAncestor {
   private static final Logger LOG = LogManager.getLogger();
-  private final RecentChainData recentChainData;
+  private static final int DEFAULT_MAX_ATTEMPTS = 5;
+  static final UInt64 BLOCK_COUNT_PER_ATTEMPT = UInt64.valueOf(10);
+  static final UInt64 SLOTS_TO_JUMP_BACK_EXPONENTIAL_BASE = UInt64.valueOf(100);
+  // We will try to find a common block downloading 10 blocks per attempt.
+  // We will try 5 times, each time jumping back 100 * 2^attempt slots.
+  // Which means we will jump back 100 slots, then 200 from previous 100, then 400, then 800, then
+  // 1600 slots.
+  // So max attempted distance from initial slot will be 3100.
+  // We will download max 50 blocks in total, before defaulting to firstNonFinalSlot
 
-  static final UInt64 OPTIMISTIC_HISTORY_LENGTH = UInt64.valueOf(3000);
-  // prysm allows a maximum range of 1000 blocks (endSlot - startSlot) due to database limitations
-  static final UInt64 BLOCK_COUNT = UInt64.valueOf(100);
+  private final RecentChainData recentChainData;
+  private final int maxAttempts;
 
   public CommonAncestor(final RecentChainData recentChainData) {
     this.recentChainData = recentChainData;
+    this.maxAttempts = DEFAULT_MAX_ATTEMPTS;
+  }
```

Comment:
should probably call the other constructor with default_max_attempts here

Reply:
fair :)

Reply:
well, on a second pass, I don't want to leak the config outside the class.. exposing only the max attempts seems inconsistent, and I should expose all of them, and I'd avoid that :)